ocw/lib/provider.py

```diff
@@ -62,5 +63,10 @@ def log_dbg(self,  message: str, *args: object):
             message = message.format(*args)
         self.logger.debug("[%s] %s", self._namespace, message)
 
-    def cmd_exec(self, cmd):
-        return subprocess.call(cmd.split())
+    def cmd_exec(self, cmd, aditional_env={}, expected_returncode = 0):
+        env = os.environ | aditional_env
+        res = subprocess.run(cmd.split(), env=env, capture_output=True)
```

Comment:
For anything related to splitting shell commands, it's better and safer to use the `shlex` module.

```suggestion
        res = subprocess.run(shlex.split(cmd), env=env, capture_output=True)
```